### ISSUE TITLE
connect-js-viz cannot be browserified

### ISSUE THREAD

**Issue by jhogendorn** [getconnect/connect-js#63]
I'm running connect-js-viz through browserify with other modules to produce a `vendor.js`. This produces these errors:

`Bundler  'vendor.js' Failed:
           Error: module not found: "connect-js-c3" from file /Us                ers/jhogendorn/Code/project/node_modules/connect-js-viz/js/c3.js`

It throws the same error with d3 as well. These both do `try catch` requires. The error is persistent if I remove the try catch blocks, and is persistent if I use remove-try-require transforms. I've honestly no idea why these packages are unable to be resolved.

Node 6.1.0
NPM 3.8.6

**Comment by AdamPD** [comment]
@jhogendorn - do you have a basic sample you could send through so we can reproduce the error?  That would give us a good running start.

**Comment by jhogendorn** [comment]
https://github.com/jhogendorn/connect-js-bug-63

**Comment by jhogendorn** [comment]
any word on this lads?

**Comment by AdamPD** [comment]
@jhogendorn - we've released a new version of connect-js (0.5.4).  I believe it was something that was put in place to deal with old Node's nested node_modules issue, which should not be needed anymore.

Could you give it a try and let me know?  I was able to have browserify build it on my end.

**Comment by jhogendorn** [comment]
I've bumped to the latest version, 0.5.4 and still have the same issue. I've confirmed this in both the main project and the example repository i provided.

**Comment by AdamPD** [comment]
That's odd - I've bumped your example repo to 0.5.4 and don't get the same issue.  I do get an issue with the gulpfile:

``` C:\OS\connect-js-bug-63\node_modules\vinyl-buffer\index.js:12
    if (file.isNull()) return push(file, next)
             ^

TypeError: file.isNull is not a function
    at DestroyableTransform.write [as _transform] (C:\OS\connect-js-bug-63\node_modules\vinyl-buffer\index.js:12:14)
    at DestroyableTransform.Transform._read (C:\OS\connect-js-bug-63\node_modules\vinyl-buffer\node_modules\readable-stream\lib\_stream_transform.js:184:10)
    at DestroyableTransform.Transform._write (C:\OS\connect-js-bug-63\node_modules\vinyl-buffer\node_modules\readable-stream\lib\_stream_transform.js:172:12)
    at doWrite (C:\OS\connect-js-bug-63\node_modules\vinyl-buffer\node_modules\readable-stream\lib\_stream_writable.js:237:10)
    at writeOrBuffer (C:\OS\connect-js-bug-63\node_modules\vinyl-buffer\node_modules\readable-stream\lib\_stream_writable.js:227:5)
    at DestroyableTransform.Writable.write (C:\OS\connect-js-bug-63\node_modules\vinyl-buffer\node_modules\readable-stream\lib\_stream_writable.js:194:11)
    at Readable.ondata (C:\OS\connect-js-bug-63\node_modules\read-only-stream\node_modules\readable-stream\lib\_stream_readable.js:543:20)
    at emitOne (events.js:90:13)
    at Readable.emit (events.js:182:7)
    at readableAddChunk (C:\OS\connect-js-bug-63\node_modules\read-only-stream\node_modules\readable-stream\lib\_stream_readable.js:210:18)
```

Is that what you're getting?

**Comment by jhogendorn** [comment]
thats an issue introduced by npm update being buggy.

you need to edit package.json, rm -rf node_modules, npm install.

**Comment by AdamPD** [comment]
I've cleared node_modules and done a fresh npm install but still getting the same issue above (as opposed to the original issue).  Are you still getting the same error message from the original issue?

**Comment by jhogendorn** [comment]
I get the original error message. a fresh npm install resolved that issue with vinyl-buffer for me.

**Comment by AdamPD** [comment]
The changes I made to the repo was to remove npm-shrinkwrap (as it was pointing to a private NPM repo I believe) and modify package.json to update both Connect packages to 0.5.4.

I'm using Node 5.7.0 to test also.

**Comment by jhogendorn** [comment]
as is stated in the reproduction repository, you will need to use nvm to load up the correct versions of node and npm to reproduce.

**Comment by AdamPD** [comment]
Even with 6.1.0 (as specified in the repo), I get the same issue with gulp (vinyl_buffer).  Very puzzling.  If I revert back to 0.5.3 for Connect, I get the original error again.

**Comment by AdamPD** [comment]
Quick thought - have you explicitly specified 0.5.4 in the package.json?

```
  "dependencies": {
    "connect-js": "0.5.4",
    "connect-js-viz": "0.5.4"
  },
```

**Comment by jhogendorn** [comment]
I have, and have now committed and pushed that change in the reproduction repo.

**Comment by AdamPD** [comment]
I've tried it on both OSX and Windows with the same result.  Are you able to delete the npm-shrinkwrap and try it again?  That's the only difference that I can see.

**Comment by jhogendorn** [comment]
I deleted the shrinkwrap and hit that vinyl error again and took a closer gander. It's not a npm install issue like I had surmised earlier (my bad, quick to blame npm, bugs out me regularly). It was an improper usage of vinyl buffer in the code after looking at my actual implementation closer. once I fixed that, dropped the shrinkwrap and clean installed my packages, it bundled fine.

So it appears your fix is good 😃  👍 

I've update the example repo, for your interest.
